python/cuml/feature_extraction/_vectorizers.py

```diff
@@ -594,23 +671,286 @@ def inverse_transform(self, X):
+        ):
+            raise ValueError(f"cudf.Series([str]) expected ,got {type(X)}")
+        self._warn_for_unused_params()
+        self._validate_params()
+        return self
+
+    def _preprocess(self, raw_documents):
+        preprocess = self.build_preprocessor()
+        return preprocess(raw_documents)
+
+    def _count_hash(self, tokenized_df):
+        """
+        Count occurrences of tokens in each document.
+        """
+        # Transform string tokens into token indexes from 0 to n_features
+        tokenized_df["token"] = tokenized_df["token"].hash_values()
+        if self.alternate_sign:
+            # below logic is equivalent to: value *= ((h >= 0) * 2) - 1
+            tokenized_df["value"] = ((tokenized_df["token"] >= 0) * 2) - 1
+            tokenized_df["token"] = tokenized_df["token"].abs() %\
+                self.n_features
+            count_ser = tokenized_df.groupby(["doc_id", "token"]).value.sum()
+            count_ser.name = "count"
+        else:
+            tokenized_df["token"] = tokenized_df["token"].abs() %\
+                self.n_features
+            count_ser = tokenized_df.groupby(["doc_id", "token"]).size()
+            count_ser.name = "count"
+
+        count_df = count_ser.reset_index(drop=False)
+        del count_ser, tokenized_df
+        return count_df
+
+    def fit_transform(self, X, y=None):
+        """
+        Transform a sequence of documents to a document-term matrix.
+
+        Parameters
+        ----------
+        X : iterable over raw text documents, length = n_samples
+            Samples. Each sample must be a text document (either bytes or
+            unicode strings, file name or file object depending on the
+            constructor argument) which will be tokenized and hashed.
+        y : any
+            Ignored. This parameter exists only for compatibility with
+            sklearn.pipeline.Pipeline.
+
+        Returns
+        -------
+        X : sparse matrix of shape (n_samples, n_features)
+            Document-term matrix.
+        """
+        return self.fit(X, y).transform(X)
+
+    def transform(self, raw_documents):
+        """
+        Transform documents to document-term matrix.
+
+        Extract token counts out of raw text documents using the vocabulary
+        fitted with fit or the one provided to the constructor.
+
+        Parameters
+        ----------
+        raw_documents : cudf.Series
+            A Series of string documents
+
+        Returns
+        -------
+        X : cupy csr array of shape (n_samples, n_features)
+            Document-term matrix.
+        """
+        docs = self._preprocess(raw_documents)
+        del raw_documents
+        n_doc = len(docs)
+        tokenized_df = self._create_tokenized_df(docs)
+        del docs
+        count_df = self._count_hash(tokenized_df)
+        del tokenized_df
+        empty_doc_ids = self._compute_empty_doc_ids(count_df, n_doc)
+        X = self._create_csr_matrix_from_count_df(
```

Comment:
I wonder if this function (`_create_csr_matrix_from_count_df`) might be useful for other things in the future, was thinking maybe it'd be better in a more general location if so. What do you think?

Reply:
Yeah, agreed. I think this should be useful for other things in the future. 

I initially wrote this for some internal/customer workflows and ended up using it across a  bunch of places including at[ tpcx-bb ](https://github.com/rapidsai/tpcx-bb/blob/69f142af4afd3c7af24f757e3e262ecaf58d2f52/tpcx_bb/xbb_tools/text_vectorizers/dist_hashing_vectorizer.py#L152-L180). 

It would be really nice to just import it from `cuml`  and not maintain that code everywhere. :-D. 

Can you point me to util like the location in cuml where I can put this in along with a test for it?

Reply:
I think a nee file called `sparse_utils` in cuml.common (i.e. here https://github.com/rapidsai/cuml/tree/branch-0.15/python/cuml/common). The one called `sparsefuncs.py` is kind of the same, but the naming is not consistent with the rest of the codebase, so you could put it there and we can rename that file to `sparse_utils.py` in a follow up PR.

Reply:
Thanks, Resolved by https://github.com/rapidsai/cuml/pull/2554/commits/c8de69317946cf5c6f8b3e090e40fd87a62b2150